### PR TITLE
263 room filters

### LINKED ISSUE #263: Find Rooms based on Filters
### User story
As a user, I can find different resources based on specific filters
### Narrative
Rooms are differentiated by details such as room type, available booking time and booking recurrence availability, and a user should have the option to filter them by these fields.
### Out of scope
Booking the requests once they are filtered.
### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a logged in authenticated user|I want to find a resource based on its type, available booking time or booking recurrence availability |I can use the various filters to increase my search accuracy and get better results |

### LINKED ISSUE #330: Room Detailed View
### User story
As a user I wanna view the details and attributes of a room
### Narrative
In the rooms table, there are attribute fields such as food/alcohol permission, capacity etc.. that are not directly shown. A user should be able to see these details
### Out of scope
N/A
### Acceptance criteria

| # | Given | When | Then |
|---|-------|------|------|
|1|I am a logged in user|I am searching for rooms in the rooms table|I can click on a details button to see extra paramteres of the room|

### PR DESCRIPTION
**Pull request by jonathanp16**
##### _What existing problem does the pull request solve?_

This solves filtering rooms by availabilities. Users can now enter as many date ranges as they want and filter to see if that room is available within those slots. It also added a detailed view for the room to see the extra attributes (food, alcohol etc).

image
image
image

##### _List all issues this PR resolves_
- closes #263 
- closes #330 

##### _Complete these items before requesting reviews_
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [x] Backend Implementation
- [x] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [ ] Documentation

### CODE REVIEW

**Comment by Alex-Hayeur** [comment]
https://sonarcloud.io/component_measures?id=CSU-Booking-Platform_application&metric=new_line_coverage&pullRequest=390&selected=CSU-Booking-Platform_application%3Aapp%2FHttp%2FControllers%2FRoomController.php&view=list

The missing code coverage seems to be coming from the recurrences section here.

File: resources/js/Components/Tables/RoomTable.vue [hunk 1]
```diff
@@ -28,19 +28,44 @@
       <tbody>
          <tr v-for="room in filteredRooms" :key="room.id">
             <td class="text-center lt-grey">{{room.id}}</td>
-            <td class="text-center lt-grey">{{room.name}}</td>
```
Comment (Alex-Hayeur) [inline resources/js/Components/Tables/RoomTable.vue:31]: Is there a reason we're dropping room name from the table? 
Admittedly the room names aren't super commonly known, but it still may be helpful to show them both.

Comment (jonathanp16) [reply resources/js/Components/Tables/RoomTable.vue:31]: The name was a placeholder for the type until that column was added. Can definitely add a column for it

Comment (Alex-Hayeur) [reply resources/js/Components/Tables/RoomTable.vue:31]: If it was always intended as a placeholder that's fine. 
I'm not even sure if most of the rooms have names actually.

**Review by EvanDime — APPROVED** [review]

**Review by Alex-Hayeur — APPROVED** [review]

**Review by Alex-Hayeur — APPROVED** [review]

**Review by EvanDime — APPROVED** [review]

**Pull request merged**